### ISSUE TITLE
Add cancellation support

### ISSUE THREAD

**Issue by brettcannon** [microsoft/ptvsd#1841]
... if possible.

https://code.visualstudio.com/updates/v1_39#_cancellation-support

**Comment by int19h** [comment]
Needs to be investigated to see which requests are cancelable, and create specific issues to do the work for each. 

I think the only one this currently makes sense for right now is variables. Since we can't actually interrupt Python evals, this is only applicable if we invoke them in a loop, which is the case there. 

Scopes are hardcoded, so definitely nothing to do there.

Stack trace has a loop, but there's no evals there, so it's unlikely to be worth it.

Completions fetch members in a single non-interruptible Python call, then iterate over that. The iteration is also fast, so again, probably not worth it.

**Comment by fabioz** [comment]
I agree variables is the one that stands of, but maybe we could try to experiment interrupting Python evals with https://docs.python.org/3/c-api/init.html?highlight=pythreadstate_setasyncexc#c.PyThreadState_SetAsyncExc (so, we could handle variables and completions).

**Comment by int19h** [comment]
Very interesting! Yeah, it sounds like that might apply to completions, in at least some cases.

(and we could also do "evaluate", but for some reason it's not on the list of cancelable requests...)
